mne/time_frequency/tfr.py

```diff
@@ -331,16 +339,15 @@ def cwt(X, Ws, use_fft=True, mode='same', decim=1):
                                     with Morlet wavelets
     """
     n_signals, n_times = X[:, ::decim].shape
-    n_frequencies = len(Ws)
 
     if use_fft:
-        coefs = _cwt_fft(X, Ws, mode)
+        coefs = _cwt_fft(X, Ws, mode, decim=decim)
     else:
-        coefs = _cwt_convolve(X, Ws, mode)
+        coefs = _cwt_convolve(X, Ws, mode, decim=decim)
 
-    tfrs = np.empty((n_signals, n_frequencies, n_times), dtype=np.complex)
+    tfrs = np.empty((n_signals, len(Ws), n_times), dtype=np.complex)
     for k, tfr in enumerate(coefs):
```

Comment:
Sorry I wasn't very clear. 

`coef` is a generator, not an array of values as I initially expected. Is there a benefit in separating the setting of the convolution (`_cwt_fft()`) from its computation  (`for tfr in coefs: ...`)? If so, I'll keep it like that.